### PR TITLE
Unwind auth

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Include @auth to the unwind create optimization

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at
 https://neo4j-graphql-toolbox-stage-2406.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by MacondoExpress** [comment]
Moved to draft to see if it's possible to move some parts to the CypherBuilder, I'll make it reviewable soon!

File: packages/graphql/tests/tck/directives/auth/projection.test.ts [hunk 1]
```diff
@@ -104,27 +104,27 @@ describe("Cypher Auth Projection", () => {
...
-            CREATE (this1:User)
-            SET this1.id = $this1_id
-            RETURN this1
+                WITH create_var1
+                CREATE (create_this0:\`User\`)
+                SET
+                    create_this0.id = create_var1.id
+                RETURN create_this0
             }
-            CALL apoc.util.validate(NOT ((this0.id IS NOT NULL AND this0.id = $projectionauth_param0)), \\"@neo4j/graphql/FORBIDDEN\\", [0])
```
Comment (MacondoExpress) [inline packages/graphql/tests/tck/directives/auth/projection.test.ts:117]: This seems to be previously wrong as `Allow` rules should not have any effect on the `CREATE` operation,
For reference: https://neo4j.com/docs/graphql-manual/current/auth/authorization/allow/

File: packages/graphql/tests/tck/directives/auth/arguments/roles/roles.test.ts [hunk 2]
```diff
@@ -273,24 +279,30 @@ describe("Cypher Auth Roles", () => {
...
+            CALL {
+                WITH create_var1
+                CREATE (create_this0:\`User\`)
+                SET
+                    create_this0.id = create_var1.id,
+                    create_this0.password = create_var1.password
+                WITH *
+                CALL apoc.util.validate(NOT (any(auth_var1 IN [\\"admin\\"] WHERE any(auth_var0 IN $auth.roles WHERE auth_var0 = auth_var1))), \\"@neo4j/graphql/FORBIDDEN\\", [0])
+                WITH *
+                CALL apoc.util.validate((create_var1.password IS NOT NULL AND NOT (any(auth_var1 IN [\\"super-admin\\"] WHERE any(auth_var0 IN $auth.roles WHERE auth_var0 = auth_var1)))), \\"@neo4j/graphql/FORBIDDEN\\", [0])
```
Comment (darrellwarde) [inline packages/graphql/tests/tck/directives/auth/arguments/roles/roles.test.ts:292]: Intrigued as to where the `create_var1.password IS NOT NULL` has come from given it's not in the original check from line 283?

**Review by darrellwarde — APPROVED** [review]
It looks good! I think my observations are mainly around the auth implementation as a whole rather than what has specifically happened here.

It looks like one discrepancy from the Cypher from before - worthwhile investigating.

We do need to think about where the information in that README goes long term.

File: .changeset/cool-tables-destroy.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"graphql-manual": patch
```
[the remark this answers is not in the document]
Comment (MacondoExpress) [reply .changeset/cool-tables-destroy.md:2]: The only update on the manual is removing Auth from the list of capabilities not supported by the bulk-create!

File: packages/graphql/tests/tck/directives/auth/arguments/roles/roles.test.ts (hunk 2 above)
Comment (MacondoExpress) [reply packages/graphql/tests/tck/directives/auth/arguments/roles/roles.test.ts:292]: So, this is a change needed when we are in the unwind optimization context when `auth` is applied at the field level, the previous behavior was to include `auth` only to the subquery where the field was present in the input argument. As of now, we have only one `Subquery` the `create_var1.password IS NOT NULL` is added to simulate the previous, in this way the auth validation could return `true` only if `password` is present in the input for that particular item!

File: packages/graphql/tests/integration/unwind-create/unwind-create-auth.int.test.ts [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+                    contextValue: neo4j.getContextValuesWithBookmarks(session.lastBookmarks(), { req }),
+                });
+
+                expect((gqlResult.errors as any[])[0].message).toBe("Forbidden");
+            } finally {
+                await session.close();
+            }
+        });
+
+        test("should not raise an error if a user is created without id", async () => {
```
[the remark this answers is not in the document]
Comment (MacondoExpress) [reply packages/graphql/tests/integration/unwind-create/unwind-create-auth.int.test.ts:106]: This is only for Field-level cases, I have to admit that our docs do not specify very well the behavior for this case but the current behavior was to include the `bind` mechanism only if the field was included in the request.
This is in sync with the `Allow` and `Roles` behaviors, but if you take a look at the docs this behavior is documented only for `Allow`.
https://neo4j.com/docs/graphql-manual/current/auth/authorization/allow/#_field_level_allow
https://neo4j.com/docs/graphql-manual/current/auth/authorization/bind/#_field_level_bind
While, even if is possible to achieve the same behavior with `Roles` is not described at all:
https://neo4j.com/docs/graphql-manual/current/auth/authorization/roles/

File: .changeset/cool-tables-destroy.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (darrellwarde) [reply .changeset/cool-tables-destroy.md:2]: More just, we don't bump the manual version, otherwise it gets published to npm!! 😆

**Review by darrellwarde — APPROVED** [review]

File: .changeset/cool-tables-destroy.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (MacondoExpress) [reply .changeset/cool-tables-destroy.md]: Ah sorry I need a crash course about our CI/CD 😂

**Pull request merged**